PR: https://github.com/odoo/odoo/pull/237519

From: 9be1c0c9be4b72609ab6febb2dfa6533f18bd9fc
From: yaso
Date: 2025-12-03 09:24:59

Breaking data model changes scores: del:10 + add:0, change matches:
-    use_epson_server_direct_print = fields.Boolean(
-    epson_server_direct_print_url = fields.Char(
-    epson_server_direct_print_id = fields.Char(
-    epson_server_pending_receipt_ids = fields.One2many(
-    receipt = fields.Text(string="Receipt")
-    pos_config_id = fields.Char(string="POS Config ID")
-    use_epson_server_direct_print = fields.Boolean(related='pos_config_id.use_epson_server_direct_print', readonly=False)
-    epson_server_direct_print_url = fields.Char(related='pos_config_id.epson_server_direct_print_url', readonly=True)
-    epson_server_direct_print_id = fields.Char(related='pos_config_id.epson_server_direct_print_id', string="Print ID", help="Set this ID in printer settings to secure your printing", readonly=True)

Total Changes: 55

[REV] point_of_sale: Epson Server Direct Print integration

This reverts commit 3b534458ab343f45efd1a6b82361f91cfb933808.
Epson Server Direct Print integration is removed as it will be relaced
by the LNA printing

closes odoo/odoo#237519

Signed-off-by: Yaroslav Soroko (yaso) <yaso@odoo.com>

================================= pseudo patch: =================================

--- a/addons/point_of_sale/models/__init__.py
+++ b/addons/point_of_sale/models/__init__.py
@@ -16,7 +16,6 @@ from . import digest
 from . import pos_category
 from . import pos_config
 from . import pos_order
-from . import pos_printer_pending_receipt
 from . import pos_session
 from . import product_pricelist
 from . import product_attribute

--- a/addons/point_of_sale/models/pos_config.py
+++ b/addons/point_of_sale/models/pos_config.py
@@ -199,7 +199,6 @@ class PosConfig(models.Model):
     order_edit_tracking = fields.Boolean(string="Track orders edits", help="Store edited orders in the backend", default=False)
     last_data_change = fields.Datetime(string='Last Write Date', readonly=True, compute='_compute_local_data_integrity', store=True)
     fallback_nomenclature_id = fields.Many2one('barcode.nomenclature', string="Fallback Nomenclature")
-    # Epson ePoS Printer Configuration
     epson_printer_ip = fields.Char(
         string='Epson Printer IP',
         help=(
@@ -207,49 +206,12 @@ class PosConfig(models.Model):
             "'Automatic Certificate Update' option is enabled in the printer settings."
         ),
     )
-    # Epson Server Direct Print printer configuration
-    use_epson_server_direct_print = fields.Boolean(
-        string="Use Epson Server Direct Printer",
-        help="Configure your printer to retrieve receipts from the server",
-    )
-    epson_server_direct_print_url = fields.Char(
-        string="Epson Server Direct Print URL",
-        help="This URL needs to be set in the printer's configuration to enable server printing.",
-        readonly=True,
-        store=False,
-        default=lambda self: f"{self.get_base_url()}/point_of_sale/epson_server_direct_print/{self.id}/get_receipt_from_queue",
-    )
-    epson_server_direct_print_id = fields.Char(
-        string="Epson Server Direct Print ID",
-        help="This ID needs to be set in the printer's configuration to secure server printing.",
-        readonly=True,
-        store=False,
-        default=lambda self: self.env['ir.config_parameter'].sudo().get_str('database.uuid')[:30].strip(),  # 30 is the max size in printer settings
-    )
-    epson_server_pending_receipt_ids = fields.One2many(
-        'pos.printer.pending.receipt',
-        'pos_config_id',
-        string='Epson Server Direct Print Queue',
-    )
     use_fast_payment = fields.Boolean('Fast Payment Validation', help="Enable fast payment methods to validate orders on the product screen.")
     fast_payment_method_ids = fields.Many2many(
         'pos.payment.method', string='Fast Payment Methods', compute="_compute_fast_payment_method_ids", relation='pos_payment_method_config_fast_validation_relation',
         store=True, help="These payment methods will be available for fast payment", readonly=False)
     statistics_for_current_session = fields.Json(string="Session Statistics", compute="_compute_statistics_for_session")
 
-    def add_receipt_to_print_queue(self, new_receipt):
-        """This method is called by the POS to add a receipt to the print queue.
-
-        :param new_receipt: The receipt content to print.
-        :return: A dictionary with the status of the operation.
-        """
-        self.ensure_one()
-        self.epson_server_pending_receipt_ids.create({
-            'pos_config_id': self.id,
-            'receipt': new_receipt,
-        })
-        return {"status": "success", "message": "Receipt added to print queue."}
-
     def _get_next_order_refs(self, device_identifier='0'):
         next_number = self.order_backend_seq_id._next()
         year_2_digits = str(datetime.now().year)[-2:]

--- a/addons/point_of_sale/models/pos_printer.py
+++ b/addons/point_of_sale/models/pos_printer.py
@@ -37,7 +37,6 @@ class PosPrinter(models.Model):
         default='epson_epos',
         selection=[
             ('epson_epos', 'Use an Epson printer'),
-            ('epson_server_direct_print', 'Use Epson Server Direct Print'),
         ]
     )
     product_categories_ids = fields.Many2many('pos.category', 'printer_category_rel', 'printer_id', 'category_id', string='Printed Product Categories')

--- a/addons/point_of_sale/models/pos_printer_pending_receipt.py
+++ b/None
@@ -1,12 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-from odoo import fields, models
-
-
-class PosPrinterPendingReceipt(models.TransientModel):
-    _name = 'pos.printer.pending.receipt'
-    _description = 'Point of Sale Printer Pending Receipt'
-    _transient_max_count = 30  # 30 receipts maximum for all PoS configs (will be vacuumed every 5 minutes)
-    _transient_max_hours = 0.1  # Each receipt will be kept for 6 minutes maximum (will be vacuumed every 5 minutes)
-
-    receipt = fields.Text(string="Receipt")
-    pos_config_id = fields.Char(string="POS Config ID")

--- a/addons/point_of_sale/models/res_config_settings.py
+++ b/addons/point_of_sale/models/res_config_settings.py
@@ -119,9 +119,6 @@ class ResConfigSettings(models.TransientModel):
     pos_fallback_nomenclature_id = fields.Many2one(related='pos_config_id.fallback_nomenclature_id', domain="[('id', '!=', barcode_nomenclature_id)]", readonly=False)
     group_pos_preset = fields.Boolean(string="Presets", implied_group="point_of_sale.group_pos_preset", help="Hide or show the Presets menu in the Point of Sale configuration.")
     pos_epson_printer_ip = fields.Char(related='pos_config_id.epson_printer_ip', readonly=False)
-    use_epson_server_direct_print = fields.Boolean(related='pos_config_id.use_epson_server_direct_print', readonly=False)
-    epson_server_direct_print_url = fields.Char(related='pos_config_id.epson_server_direct_print_url', readonly=True)
-    epson_server_direct_print_id = fields.Char(related='pos_config_id.epson_server_direct_print_id', string="Print ID", help="Set this ID in printer settings to secure your printing", readonly=True)
     pos_use_fast_payment = fields.Boolean(related='pos_config_id.use_fast_payment', readonly=False)
     pos_fast_payment_method_ids = fields.Many2many(related='pos_config_id.fast_payment_method_ids', readonly=False)
 
